### PR TITLE
HTML search: adjustments to type-dependent CSS classnames and defaults

### PR DESCRIPTION
**Pull request by jayaddison**
### Feature or Bugfix
- Bugfix
- Refactoring

### Purpose
- This is a follow-up to #12474 to resolve [some](https://github.com/sphinx-doc/sphinx/pull/12474#issuecomment-2294873020) [quirks](https://github.com/sphinx-doc/sphinx/pull/12474#issuecomment-2306687131) observed there.
- Removes the default 'unicode white square' list item styling (in the built-in Sphinx13 theme) when no type-dependent information is found on search results (as is the case from RTD-based hosted search results).
- Renames the type-dependent CSS classname indicator prefix from `context-` to `kind-` -- because we already use the term `context` elsewhere to refer to search summary snippets (and that is therefore a widely-deployed CSS classname).

### Detail
- Rename `context-` prefixes to `kind-`.
- Remove the default Unicode `25A1` codepoint `list-style-type` for default search results (in the built-in `sphinx13` theme).

### Relates
- Follow-up to #12474.

cc @AA-Turner @timhoffm

### CODE REVIEW

**Comment by timhoffm** [comment]
We also need to update the docs: https://www.sphinx-doc.org/en/master/development/html_themes/index.html#styling-with-css

In addition to the `kind` renaming, I suggest to add a note that this is only valid for the static builtin search.

Also, since sphinx13 is only used for the sphinx docs and they are deployed via rtd, I suggest to remove all the CSS added to it in #12474, because it does not have any effect.

**Comment by timhoffm** [comment]
This is release-critical: The `context` naming has not been published yet. By renaming before a release, we prevent trouble with having the `context` naming in a public release.

**Comment by jayaddison** [comment]
> We also need to update the docs: https://www.sphinx-doc.org/en/master/development/html_themes/index.html#styling-with-css
>
> In addition to the kind renaming, I suggest to add a note that this is only valid for the static builtin search.

Makes sense, yep - I'll add a note about that soon.

> Also, since sphinx13 is only used for the sphinx docs and they are deployed via rtd, I suggest to remove all the CSS added to it in #12474, because it does not have any effect.

I'm not sure I agree with this; I fairly frequently (not always, by any means) build Sphinx's documentation locally during development and testing, and it's useful to have the functinality available there to check and inspect.

**Comment by timhoffm** [comment]
> I'm not sure I agree with this; I fairly frequently (not always, by any means) build Sphinx's documentation locally during development and testing, and it's useful to have the functinality available there to check and inspect.

Sure. In that case, add a comment that this is only for dev builds to avoid confusion for people looking into the code and trying to map it to released docs.

File: doc/_themes/sphinx13/static/sphinx13.css [hunk 1]
```diff
@@ -702,17 +702,18 @@ ul.search {
 }
 ul.search li {
     padding: 5px 0 5px 10px;
-    list-style-type: "\25A1";  /* Unicode: White Square */
 }
-ul.search li.context-index {
+
+/* note: these rules apply to search results from the built-in Sphinx HTML/JS search engine */
```
Comment (timhoffm) [inline doc/_themes/sphinx13/static/sphinx13.css]:
```suggestion
/* note: these rules apply to search results from the built-in Sphinx HTML/JS search engine
   and only take effect in dev builds. The released docs use the ReadTheDocs search engine and are not affected. */
```

**Comment by jayaddison** [comment]
@AA-Turner @picnixz could we fit this in before a v8.0.3 release?  I had suggested a CSS naming convention in #12474 that overlaps with some already-in-use CSS classes; it'd be nice to avoid these appearing in a published release (an `HTTP Referer`-type situation).

**Comment by AA-Turner** [comment]
Next release will be 8.1. Do we need to make a special 8.0.3 release?

A

**Comment by jayaddison** [comment]
> Next release will be 8.1. Do we need to make a special 8.0.3 release?

No, we do not need a patch/hotfix release for this; the `context-...` CSS prefix hasn't been included in any releases so far.

Provided that we merge this before the next release (8.1 in that case), then all is well.

**Review by timhoffm — APPROVED** [review]

**Comment by jayaddison** [comment]
Thank you @AA-Turner!

**Pull request merged**